src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs

```diff
@@ -769,29 +766,43 @@ private static bool IsExecutable(string fullPath)
                 return false;
             }
 
-            Interop.Sys.Permissions permissions = (Interop.Sys.Permissions)fileinfo.Mode;
+            Interop.Sys.Permissions permissions = ((Interop.Sys.Permissions)fileinfo.Mode) & Interop.Sys.Permissions.S_IXUGO;
 
-            if (s_euid == 0)
+            // Avoid checking user/group when everyone has the same x-permission (common).
+            if (permissions == Interop.Sys.Permissions.S_IXUGO)
             {
-                // We're root.
-                return (permissions & Interop.Sys.Permissions.S_IXUGO) != 0;
+                return true;
+            }
+            else if (permissions == 0)
+            {
+                return false;
             }
 
-            if (s_euid == fileinfo.Uid)
+            uint euid = Interop.Sys.GetEUid();
+            if (euid == 0)
+            {
+                // We're root.
+                return true;
```

Comment:
a very minor nit: we can put such short comments in the same line to save some space

```suggestion
                return true; // We're root.
```